packages/mdc-textfield/_mixins.scss

```diff
@@ -62,3 +62,263 @@
+    }
+  }
+
+  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-text-field__input,
+  .mdc-text-field__label,
+  + .mdc-text-field-helper-text {
+    @include mdc-theme-prop(color, text-disabled-on-light);
+  }
+  // stylelint-enable plugin/selector-bem-pattern
+
+  .mdc-text-field__input,
+  .mdc-text-field__label {
+    @include mdc-theme-dark(".mdc-text-field") {
+      @include mdc-theme-prop(color, text-disabled-on-dark);
+    }
+  }
+
+  @include mdc-theme-dark(".mdc-text-field", true) {
+    // stylelint-disable plugin/selector-bem-pattern
+    + .mdc-text-field-helper-text {
+      @include mdc-theme-prop(color, text-disabled-on-dark);
+    }
+    // stylelint-enable plugin/selector-bem-pattern
+  }
+
+  .mdc-text-field__label {
+    bottom: 8px;
+    cursor: default;
+  }
+}
+
+@mixin mdc-text-field-focused_ {
+  .mdc-text-field__input:required + .mdc-text-field__label::after {
+    color: $mdc-text-field-error-on-light;
+
+    @include mdc-theme-dark(".mdc-text-field", true) {
+      color: $mdc-text-field-error-on-dark;
+    }
+  }
+}
+
+// outline
+
+@mixin mdc-text-field-focused-outline_() {
+  .mdc-text-field__idle-outline,
+  .mdc-text-field__label--float-above ~ .mdc-text-field__idle-outline {
+    opacity: 0;
+  }
+
+  .mdc-text-field__outline,
+  .mdc-text-field__label--float-above ~ .mdc-text-field__outline {
+    opacity: 1;
+  }
+
+  .mdc-text-field__outline-path {
+    @include mdc-theme-prop(stroke, primary);
+
+    stroke-width: 2px;
+  }
+}
+
+@mixin mdc-text-field-focused-disabled_() {
+  @include mdc-theme-prop(color, $mdc-text-field-light-placeholder);
+
+  .mdc-text-field__input {
+    border-bottom: none;
+  }
+
+  .mdc-text-field__outline-path {
+    stroke: $mdc-text-field-outlined-disabled-border;
+    stroke-width: 1px;
+  }
+
+  .mdc-text-field__idle-outline {
+    border-color: $mdc-text-field-outlined-disabled-border;
+  }
+
+  .mdc-text-field__label {
+    bottom: 20px;
+  }
+
+  .mdc-text-field__icon {
+    @include mdc-theme-prop(color, $mdc-text-field-disabled-icon-on-light);
+  }
+}
+
+// textarea
+@mixin mdc-text-field-focused-textarea_ {
+  @include mdc-theme-prop(border-color, primary);
+}
+
+@mixin mdc-text-field-invalid-textarea_ {
+  border-color: $mdc-text-field-error-on-light;
+}
+
+@mixin mdc-text-field-disabled-textarea_ {
+  border-style: solid;
+  border-color: $mdc-text-field-disabled-border-on-light;
+  background-color: $mdc-textarea-disabled-background-on-light;
+
+  @include mdc-theme-dark(".mdc-text-field") {
+    border-color: $mdc-text-field-disabled-border-on-dark;
+    background-color: $mdc-textarea-disabled-background-on-dark;
+  }
+
+  .mdc-text-field__label {
+    background-color: $mdc-textarea-disabled-background-on-light;
+
+    @include mdc-theme-dark(".mdc-text-field") {
+      background-color: $mdc-textarea-disabled-background-on-dark;
+    }
+  }
+}
+
+// box
+@mixin mdc-text-field-box-disabled_ {
```

Comment:
Fix consistency: mixins above are named in order of state-variant, but this one is named variant-state.
(I actually prefer the variant-state naming more, but up to you as long as it's consistent.)

Reply:
I also like variant-state...changing